### PR TITLE
Feature/page variables on cat archive

### LINKED ISSUE #1647: Parse `{page_uri}` and `{page_url}` within Channel Category Archive
- [x] I am capable and would like to work on implementation of this feature if it is considered.

### PR DESCRIPTION
**Pull request by robsonsobral**
## Overview

Parse `{page_url}` and `{page_uri}` on Channel Category Archive

Resolves [#1647](https://github.com/ExpressionEngine/ExpressionEngine/issues/1647).

## Nature of This Change

- [ ] 🐛 Fixes a bug
- [x] 🚀 Implements a new feature
- [ ] 🛁 Refactors existing code
- [ ] 💅 Fixes coding style
- [ ] ✅ Adds tests
- [ ] 👽 Adds new dependency
- [ ] 🔥 Removes unused files / code
- [ ] 🔒 Improves security 

## Is this backwards compatible?

- [x] Yes
- [ ] No

## Documentation

User Guide Pull Request: https://github.com/ExpressionEngine/ExpressionEngine-User-Guide/pull/336

### CODE REVIEW

**Review by intoeetive — CHANGES_REQUESTED** [review]
@robsonsobral would you mind pulling in latest changes from `6.2.0` branch - those are approved to be included into 6.2.0 release and contain some changes to this file. In particular, it is replacing `swap_var_single` function in the block you're working with here with `parse_variables_row`.  `swap_var_single` is marked as legacy, and while it's not deprecated, it's recommended to use it less.

Currently I have a conflict when trying to merge 6.2.0 into your branch, so if you could resolve it that would be great!

**Comment by robsonsobral** [comment]
Sure, boss! I still find weird you all merging back changes to PRs. Who opened the PR should be responsible to do that.

Let's do it!

**Comment by robsonsobral** [comment]
Rebased.

File: system/ee/ExpressionEngine/Addons/channel/mod.channel.php
Comment (intoeetive) [inline system/ee/ExpressionEngine/Addons/channel/mod.channel.php]: I think depending on error reporting level in PHP using this without conditional check might give a notice if the entry does not have corresponding page.

File: system/ee/ExpressionEngine/Addons/channel/mod.channel.php
Comment (intoeetive) [inline system/ee/ExpressionEngine/Addons/channel/mod.channel.php:3207]: I am not sure I understand the reason for this chunk of code. I think `$pages = ee()->config->site_pages($site_id);` should be enough?

File: system/ee/ExpressionEngine/Addons/channel/mod.channel.php
Comment (robsonsobral) [reply system/ee/ExpressionEngine/Addons/channel/mod.channel.php:3214]: And if I tell you this is a plain copy/paste from parser lib? Let's fix it!

File: system/ee/ExpressionEngine/Addons/channel/mod.channel.php
Comment (robsonsobral) [reply system/ee/ExpressionEngine/Addons/channel/mod.channel.php:3207]: And if I tell you this is a plain copy/paste from parser lib? Let's fix it!

Comment (intoeetive) [reply system/ee/ExpressionEngine/Addons/channel/mod.channel.php:3207]: That is interesting! I'll need to figure out why this code was build exactly this way - and then hopefully improve it

Comment (robsonsobral) [reply system/ee/ExpressionEngine/Addons/channel/mod.channel.php:3207]: From [here](https://github.com/ExpressionEngine/ExpressionEngine/blob/292951c13a4e6a1605d08b7aac1e86641e39e9b9/system/ee/legacy/libraries/channel_entries_parser/Parser.php#L164), @intoeetive .

Comment (robsonsobral) [reply system/ee/ExpressionEngine/Addons/channel/mod.channel.php:3207]: I've found the reason, @intoeetive . `config_item('site_pages');` returns the pages only for the current site. The loop adds pages from a site or sites declared on a possible `site` parameter.

So, it's right.

**Comment by robsonsobral** [comment]
Hey, @intoeetive . My PR was rejected or closed by mistake? 🙁

**Comment by intoeetive** [comment]
Looks like it was closed automatically because the target branch was removed automatically. Reopening now

**Comment by robsonsobral** [comment]
Thank you! So, what I need to do, @intoeetive ?

**Comment by intoeetive** [comment]
Wait until we review (again) and merge it, @robsonsobral :)

**Review by intoeetive — APPROVED** [review]
Looks good to me.

**Review by TomJaeger — APPROVED** [review]

**Pull request merged**
